[Issue 15] POST /notebooks/* API

dalogsdon opened the pull request:
Also includes error handling updates.

:chicken:

Refs #15

parente commented:
LGTM but get another opinion from @jhpedemonte who's been living here longer than me.

jhpedemonte commented on routes/index.js:
```diff
@@ -2,12 +2,23 @@
+var upload = multer({
+    storage: storage
+});
+
```
I would prefer that this code chunk (and require of `multer`, etc) be moved to ` app/upload-notebook.js`. Then in this file, we just require that file and set the `upload` object, to be used below. Makes the code cleaner and more modular.

dalogsdon replied:
Yeah this is a leftover from some refactor. I'll move it over as suggested.

jhpedemonte commented:
Also, this is a candidate for integration test:

1. Upload notebook with 'phrase A'.
2. Query URL for that notebook and check that response has 'phrase A'.
3. Update notebook, now with 'phrase B'.
3. Query URL for notebook and check that response now has 'phrase B'.

dalogsdon commented:
Updated as per @jhpedemonte's comments.

Moved `multer` usage into the `upload-notebooks` module and did some basic form limit and file extension validation.
